python/dolfinx/mesh.py

```diff
@@ -333,6 +334,37 @@ def refine(
     return Mesh(mesh1, mesh._ufl_domain)
 
 
+def refine_interval(
+    mesh: Mesh,
+    cells: typing.Optional[np.ndarray] = None,
+    redistribute: bool = True,
+    ghost_mode: GhostMode = GhostMode.none,
+) -> tuple[Mesh, npt.NDArray[np.int32]]:
+    """Refine a (topologically) one dimensional mesh.
+
+    Args:
+        mesh: Mesh to refine
+        cells: Indices of cells, i.e. edges, to split druing refinement. If ``None``, mesh
+            refinement is uniform.
+        redistribute: Refined mesh is re-partitioned if ``True``.
+        ghost_mode: ghost mode of the refined mesh
```

Comment:
What is the use case for having a different ghost mode on the refined mesh?

Reply:
Thats a good question. Would require that the refined mesh has different forms being evaluated on it, which do not require the same ghost mode (or the other way around) - not sure of a concrete example though. Maybe when coupling a surrogate model into an adaptively refined approach?